### PR TITLE
Feature/insert update

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

TODOs:
- [x] Fix tests on DBIterator
- [x] Add tests on new methods
- [x] Add documentation

### CODE REVIEW

File: inc/certificate.class.php [hunk 1]
```diff
@@ -689,9 +689,6 @@ static function cronCertificate($task = null) {
       }
 
       $message      = [];
-      $items_notice = [];
```
Comment (orthagh) [inline inc/certificate.class.php]: seems strange, but i checked, and it's ok

File: inc/log.class.php [hunk 2]
```diff
@@ -228,15 +228,20 @@ static function history ($items_id, $itemtype, $changes, $itemtype_link = '', $l
...
-                        '".addslashes($username)."', '$date_mod', '$id_search_option',
-                        '$old_value', '$new_value')";
-
-      if ($DB->query($query)) {
+      $params = [
+         'items_id'          => $items_id,
+         'itemtype'          => $itemtype,
+         'itemtype_link'     => $itemtype_link,
+         'linked_action'     => $linked_action,
+         'user_name'         => $username,
```
Comment (orthagh) [inline inc/log.class.php]: addslashes dropped ? you kept one here: https://github.com/glpi-project/glpi/pull/3250/files#diff-11a09748e657af84b23fd2599f1f3101R1248

Comment (trasher) [reply inc/log.class.php]: Indeed; this must stay

File: inc/certificate.class.php (hunk 1 above)
Comment (trasher) [reply inc/certificate.class.php]: Yep; they were not used

**Comment by trasher** [comment]
Doc: https://github.com/glpi-project/docdev/pull/71

**Pull request merged**